src/components/action-popover/action-popover.test.tsx

```diff
@@ -0,0 +1,2303 @@
+    );
+  }).toThrow(
+    "ActionPopover only accepts children of type `ActionPopoverItem`" +
+      " and `ActionPopoverDivider`."
+  );
+
+  globalConsoleSpy.mockRestore();
+});
+
+describe("when an item has a submenu with default (left) alignment", () => {
+  it("renders the appropriate icon with the correct alignment", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <ActionPopover>
+        <ActionPopoverItem>example item 1</ActionPopoverItem>
+        <ActionPopoverItem>example item 2</ActionPopoverItem>
+        <ActionPopoverItem
+          submenu={
+            <ActionPopoverMenu>
+              <ActionPopoverItem>submenu item 1</ActionPopoverItem>
+              <ActionPopoverItem>submenu item 2</ActionPopoverItem>
+            </ActionPopoverMenu>
+          }
+        >
+          example item with submenu
+        </ActionPopoverItem>
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    // there are 2 elements with data-role="icon", the other being the main menu button.
+    // Unfortunately there seems to be no way with the RTL options for getByTestId to select the correct one, so we have to index
+    // into the array of all such elements
+    const submenuIcon = screen.getAllByTestId("icon")[1];
+    expect(submenuIcon).toHaveStyleRule(
+      "content",
+      `"${iconUnicodes.chevron_left_thick}"`,
+      { modifier: "&::before" }
+    );
+    expect(submenuIcon).toHaveStyle({ left: "-5px" });
+  });
+
+  it("opens the submenu on mouseenter", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    render(
+      <ActionPopover>
+        <ActionPopoverItem>example item 1</ActionPopoverItem>
+        <ActionPopoverItem>example item 2</ActionPopoverItem>
+        <ActionPopoverItem
+          submenu={
+            <ActionPopoverMenu>
+              <ActionPopoverItem>submenu item 1</ActionPopoverItem>
+              <ActionPopoverItem>submenu item 2</ActionPopoverItem>
+            </ActionPopoverMenu>
+          }
+        >
+          example item with submenu
+        </ActionPopoverItem>
+      </ActionPopover>
+    );
+
+    await user.click(screen.getByRole("button"));
+
+    expect(
+      screen.queryByRole("button", { name: "submenu item 1" })
+    ).not.toBeInTheDocument();
+
+    // move mouse over the item with the submenu
+    await user.pointer({
+      target: screen.getByRole("button", { name: "example item with submenu" }),
+    });
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(
+      screen.getByRole("button", { name: "submenu item 1" })
+    ).toBeInTheDocument();
+  });
+
+  // test needed for coverage of the clearTimeout call - it's not clear if the clearTimeout is actually needed
+  // and if it has any real observable consequences. We should investigate that and remove the call, and this
+  // test, if it is not needed.
```

Comment:
comment: let's make a ticket a for this task so it doesn't get lost

Reply:
now created as FE-6722